src/charm.py

```diff
@@ -176,13 +122,71 @@ def _run_sql_migration(self) -> None:
 
     def _on_pebble_ready(self, event) -> None:
         """Event Handler for pebble ready event."""
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Kratos container. Deferring event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Kratos container")
+            return
+
         self.unit.status = MaintenanceStatus("Configuring/deploying resources")
-        self._update_container(event)
 
-    def _on_database_changed(self, event: DatabaseCreatedEvent) -> None:
+        with open("src/identity.default.schema.json", encoding="utf-8") as schema_file:
+            schema = schema_file.read()
+            self._container.push(self._identity_schema_file_path, schema, make_dirs=True)
+        self._container.add_layer(self._container_name, self._pebble_layer, combine=True)
+        logger.info("Pebble plan updated with new configuration, replanning")
+        self._container.replan()
+        if self.database.is_database_created():
+            self._container.push(self._config_file_path, self._config, make_dirs=True)
+            self._container.start(self._container_name)
```

Comment:
I set the service to startup disabled because it depends on a valid config file which might not be available at the time of the event. This snippet is added to make sure the container can recover if it was terminated unexpectedly (see earlier comments from Natalia)